pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1008,6 +1008,8 @@ func (m *ManagerImpl) callGetPreferredAllocationIfAvailable(podUID, contName, re
 // the allocated capacity. This allows pods that have already been scheduled on
 // the node to pass GeneralPredicates admission checking even upon device plugin failure.
 func (m *ManagerImpl) sanitizeNodeAllocatable(node *schedulerframework.NodeInfo) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
```

Comment:
There should be no need to lock, because the caller has already added？

https://github.com/kubernetes/kubernetes/blob/1e26b14272fbc70939c072fa0e7e1392144e8971/pkg/kubelet/cm/devicemanager/manager.go#L393-L406

Reply:
Thanks for your feedback @silenceper

Made a mistake by simply taking the changes from #99370 as granted.

Will remove the mutex for `sanitizeNodeAllocatable` and submit again.

Reply:
we also need add lock for this function:
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/kubelet/cm/devicemanager/topology_hints.go#L153-L197

Reply:
Good catch! I will add that lock!

I think we should also add lock to following function:
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/kubelet/cm/devicemanager/topology_hints.go#L141-L144

Reply:
yes, you are right.

Reply:
as well as deviceHasTopologyAlignment since allDevices is also just another map
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/kubelet/cm/devicemanager/topology_hints.go#L131-L138

Reply:
@dimakow 

I think do not need add lock for `getAvailableDevices`，we should add lock for this content:

`m.Lock()`
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/kubelet/cm/devicemanager/topology_hints.go#L43-L66
`m.Unlock()`


because `deviceHasTopologyAlignment` also need add lock, but the caller already add lock like `devicesToAllocate`.

Reply:
Ok, i follow that thought. However it seems to me that lock handling (at least) in device manager is worth improving by some abstraction layers for the maps of manager.go so that not each caller has to lock again...but that is probably worth another PR

Reply:
Also it seems that in latest master we have a new function in topology_hints.go `GetPodTopologyHints` which would need same lock we added here. But that is also something worth another PR imho.

Reply:
@silenceper

Your proposed logic would leave the mutex locked forever in case of continue statements, no? So it would be best to introduce locks in the specific functions as proposed earlier. However since the locking in device-manager in general needs some overhaul i think it worth another PR to master (and probably backport).

Thus i would propose to move ahead with this PR and fix the issue it was meant to fix.